From 658040ba61ff5901a8122d0cb9713f358530276e Mon Sep 17 00:00:00 2001
From: Zhixiong Lin <zhixiong.lin@rock-chips.com>
Date: Fri, 20 Nov 2020 16:36:14 +0800
Subject: [PATCH 227/297] bootanimation: set the attrib_list format to
 RGBx_8888.

Especially for 3288 device with Bifrost DDK of r12 and later.
Bifrost DDK r12 supports an new egl_config with native format of
RGBA_1010102, bootanim before this commit would use that egl_config.
While, rk_vop does not support to process a graphics buffer with that
format, hwc_composition could not be used, resulting in poor performance.

For the attrib_list modified, please refer to egl_1.4_spec, "3.4.1.2
Sorting of EGLConfigs", "3. Special: ".

Signed-off-by: chenzhen <chenzhen@rock-chips.com>
Signed-off-by: Zhixiong Lin <zhixiong.lin@rock-chips.com>
Change-Id: I435bf3a478dc414adf921da64c1187ea7b51d1ca
---
 cmds/bootanimation/BootAnimation.cpp | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/cmds/bootanimation/BootAnimation.cpp b/cmds/bootanimation/BootAnimation.cpp
index 91b14ee3a59..3abf919eda7 100644
--- a/cmds/bootanimation/BootAnimation.cpp
+++ b/cmds/bootanimation/BootAnimation.cpp
@@ -338,10 +338,11 @@ public:
 
 EGLConfig BootAnimation::getEglConfig(const EGLDisplay& display) {
     const EGLint attribs[] = {
-        EGL_RED_SIZE,   8,
-        EGL_GREEN_SIZE, 8,
-        EGL_BLUE_SIZE,  8,
+        EGL_RED_SIZE,   0,
+        EGL_GREEN_SIZE, 0,
+        EGL_BLUE_SIZE,  0,
         EGL_DEPTH_SIZE, 0,
+        EGL_BUFFER_SIZE, 24,
         EGL_NONE
     };
     EGLint numConfigs;
-- 
2.36.0

